Added form validation for AddEvent page

ryanchansf opened the pull request:
Made name, date, start time and end time required fields. Also added check to make sure that end time comes after start time. Small problem: for date, start time and end time, the error message turns from red to grey when the field is completely empty. Validation still works, but the only difference is that the color of the error message changes.

<img width="1440" alt="Screen Shot 2022-05-17 at 9 34 37 PM" src="https://user-images.githubusercontent.com/91078767/168957976-3256b5b8-7c3f-4a3e-a18b-a4a16d06455d.png">

Note: some CSS bugs still here with the bottom event card and add button extending too far down and cutting off

ryanchansf committed:
Added form validation for AddEvent page

Made name, date, start time and end time required fields. Also added check to make sure that end time comes after start time. Small problem: for date, start time and end time, the error message turns from red to grey when the field is completely empty. Validation still works, but the only difference is that the color of the error message changes.

vwinstea reviewed (changes requested):
Looks good! Thanks for working on this. There's a few things I think we should change before we merge it in though:

- The add button should remain grey when you hover over it if the event does not pass validation. Right now it turns green on hover even if it isn't valid
- If the event doesn't pass validation, there should always be an error message visible on the page. Right now I noticed that if the title or location isn't filled out, the event won't create but it also won't tell me what I'm doing wrong.
- Check the console for errors from this page (right click -> inspect elements -> console tab). Right now there are a few that we should probably be handling

ryanchansf committed:
Updated AddEvent page

Made some changes on the add button and validation functionality. I decided to get rid of the disable attribute on the add button, since that interfered with the error message popping up when trying to submit empty fields. This way, the error message would be prioritized over the coloring of the button (as it is in the Sign Up page as well). The button will now always be green, except it will only submit if the text fields are valid and if not, it will throw an error message.

ryanchansf commented:
Made some changes on the add button and validation functionality. I decided to get rid of the disable attribute on the add button, since that interfered with the error message popping up when trying to submit empty fields. This way, the error message would be prioritized over the coloring of the button (as it is in the Sign Up page as well). The button will now always be green, except it will only submit if the text fields are valid and if not, it will throw an error message.
![Screen Shot 2022-05-28 at 7 45 37 PM (2)](https://user-images.githubusercontent.com/91078767/170849755-c49965ea-8715-40bd-a881-67bf71963a13.png)

vwinstea reviewed (approved):
Works well, thanks for making those fixes!
